src/signal_manager.sh

```diff
@@ -0,0 +1,48 @@
+. "$KW_LIB_DIR/kwio.sh" --source-only
+
+declare -ga SIGNAL_MANAGER_VALID_SIGNALS=( $(trap -l | sed -E 's/[0-9]+\)/ /g') )
+
+# This functions prints the default message when interrupting kw
+function default_interrupt_handler()
+{
+  say $'\n'Oh no! An interruption! See ya...
+}
+
+# This function adds a new signal handler to an arbitrary signal
+#
+# @command The handler to be called when a signal is catched
+# @signals The signals to attach to this handler
+#
+# If command is empty, use the default handler
+# If the signal is empty, use the default signals
+
+function signal_manager()
+{
+  local command="$1"
+  shift
+  local -a signals=( "$@" )
+
+  if [[ -z "$command" ]]; then
+    command='default_interrupt_handler'
+  fi
```

Comment:
You can simplify this to `[[ -z "$command" ]] && command='default_interrupt_handler'`

Reply:
I think the original is more readable for someone not that familiar with bash